modules/i3s/src/helpers/promises-handler.js

```diff
@@ -0,0 +1,22 @@
+export const REJECTED_STATUS = 'rejected';
+export const FULFILLED_STATUS = 'fulfilled';
+
+/**
+ * Handle list of promises based on browser compatibility
+ * @param {Promise[]} promises
+ * @returns {Promise}
+ */
+export function handlePromises(promises) {
+  return Promise.allSettled ? Promise.allSettled(promises) : Promise.all(promises.map(reflect));
```

Comment:
Which browser is having issues with allSettled? Maybe mention in comment?

We could also just polyfill `allSettled`?

Reply:
`Promise.allSettled() `was implemented in `ECMAScript 2020, the 11th edition`. It has wide browser support but pretty much for newest versions of browsers:

![image](https://user-images.githubusercontent.com/70207219/115839589-77768a00-a423-11eb-9d0b-6976ea094656.png)

For example `Promise.all() ` support table:
![image](https://user-images.githubusercontent.com/70207219/115839801-b4428100-a423-11eb-85e6-78c310321fda.png)

Yes I think we can add a polyfill for it.